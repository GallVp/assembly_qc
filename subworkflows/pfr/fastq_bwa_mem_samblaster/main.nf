include { BWA_INDEX         } from '../../../modules/pfr/bwa/index/main'
include { BWA_MEM           } from '../../../modules/pfr/bwa/mem/main'
include { SAMBLASTER        } from '../../../modules/pfr/samblaster/main'

workflow FASTQ_BWA_MEM_SAMBLASTER {

    take:
    ch_fastq                // channel: [ val(meta), [ fq ] ]
    ch_fasta_bwa_index      // channel: [ fasta, index ]; fast | index

    main:
    ch_versions             = Channel.empty()

    ch_has_index            = ch_fasta_bwa_index
                            | branch { fasta, index ->
                                yes: index
                                no: !index
                            }

    // MODULE: BWA_INDEX
    BWA_INDEX ( ch_has_index.no.map { fasta, index -> [ [], fasta ] } )

    ch_bwa_index            = BWA_INDEX.out.index
                            | mix(
                                ch_has_index.yes
                                | map { fasta, index ->
                                    [ [], index ]
                                }
                            )
                            | map { dummy, index -> index }

    ch_versions             = ch_versions.mix(BWA_INDEX.out.versions.first())

    // MODULE: BWA_MEM
    ch_mem_inputs           = ch_fastq
                            | combine(
                                ch_bwa_index
                            )
    def sort_bam            = false
    BWA_MEM(
        ch_mem_inputs.map { meta, fq, index -> [ meta, fq ] },
        ch_mem_inputs.map { meta, fq, index -> [ [], index ] },
        sort_bam
    )

    ch_mem_bam              = BWA_MEM.out.bam
    ch_versions             = ch_versions.mix(BWA_MEM.out.versions.first())

    // MODULE: SAMBLASTER
    SAMBLASTER ( ch_mem_bam )

    ch_blasted_bam          = SAMBLASTER.out.bam
    ch_versions             = ch_versions.mix(SAMBLASTER.out.versions.first())

    emit:
    bam                     = SAMBLASTER.out.bam    // channel: [ val(meta), bam ]
    versions                = ch_versions           // channel: [ versions.yml ]
}
