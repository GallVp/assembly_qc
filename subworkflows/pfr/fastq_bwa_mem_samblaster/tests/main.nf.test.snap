{
    "sarscov2-fq-gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "ref_id": "genome"
                        },
                        "test.on.genome.samblaster.bam:md5,496319fc81c383a9ae6ab52592876c9b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2ed54ca5e54063cb579273d0792465a7",
                    "versions.yml:md5,49d22cda9beaf6ea1a1ad838ef4a4255",
                    "versions.yml:md5,6a2baa7f2d1d555fe604e451624f414b"
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "ref_id": "genome"
                        },
                        "test.on.genome.samblaster.bam:md5,496319fc81c383a9ae6ab52592876c9b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2ed54ca5e54063cb579273d0792465a7",
                    "versions.yml:md5,49d22cda9beaf6ea1a1ad838ef4a4255",
                    "versions.yml:md5,6a2baa7f2d1d555fe604e451624f414b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-08T19:37:23.464854"
    }
}